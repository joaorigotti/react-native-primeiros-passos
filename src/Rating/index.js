import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableOpacity,
} from 'react-native';
import { getWidth } from '../Utils';
import Title from '../Title';
import Separator from '../Separator';
import { increment, decrement } from '../Business';

class Rating extends Component {
  constructor(props) {
    super(props);
    this.state = { rating: props.defaultValue };
    this.increment = this.increment.bind(this);
    this.decrement = this.decrement.bind(this);
  }

  increment() {
    let newRating = increment(this.state.rating);
    this.setState(() => ({ rating: parseInt(newRating) }));
  }

  decrement() {
    let newRating = decrement(this.state.rating);
    this.setState(() => ({ rating: parseInt(newRating) }));
  }

  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={this.decrement}>
          <Image source={require('../../assets/images/minus.png')} />
        </TouchableOpacity>

        <View style={styles.ratingContainer}>
          <Text style={styles.rating}>{this.state.rating}</Text>
          <Separator width={160} color="#8cc63f" />
          <Title color="#808080" size={16}>Rating</Title>
        </View>

        <TouchableOpacity onPress={this.increment}>
          <Image source={require('../../assets/images/plus.png')} />
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    width: getWidth(),
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center'
  },

  ratingContainer: {
    width: getWidth() / 3,
    alignItems: 'center'
  },

  rating: {
    fontFamily: 'RobotoSlab-Regular',
    fontSize: 56,
    fontWeight: 'bold',
    color: '#4d4d4d',
  },
});

export default Rating