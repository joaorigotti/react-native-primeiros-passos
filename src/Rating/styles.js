import { StyleSheet } from 'react-native';
import * as cs from '../CommonStyles';

export default () => StyleSheet.create({
  container: {
    width: cs.SCREEN_WIDTH,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center'
  },

  ratingContainer: {
    width: cs.SCREEN_WIDTH / 3,
    alignItems: 'center'
  },

  rating: {
    color: cs.COLOR.gray30,
    fontFamily: cs.MAIN_FONT,
    fontSize: 56,
    fontWeight: 'bold',
  },
});