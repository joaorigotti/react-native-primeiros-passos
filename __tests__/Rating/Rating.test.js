import React from 'react';
import { shallow } from 'enzyme';
import Rating from '../../src/Rating';

describe('<Rating />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Rating />
    );
    
    expect(wrapper).toMatchSnapshot();
  });
});