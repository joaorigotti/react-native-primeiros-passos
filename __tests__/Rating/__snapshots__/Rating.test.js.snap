// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "width": 750,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Image
      source={1}
    />
  </TouchableOpacity>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": 250,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#4d4d4d",
          "fontFamily": "RobotoSlab-Regular",
          "fontSize": 56,
          "fontWeight": "bold",
        }
      }
    />
    <Separator
      color="#8cc63f"
      width={160}
    />
    <Photo
      color="#808080"
      size={16}
    >
      Rating
    </Photo>
  </View>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Image
      source={1}
    />
  </TouchableOpacity>
</View>
`;
